internal/testutils/servertestutils.go

```diff
@@ -0,0 +1,785 @@
+package testutils
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/canonical/notary/internal/server"
+)
+
+func MustPrepareServer(t *testing.T) *httptest.Server {
+	t.Helper()
+
+	tempDir := t.TempDir()
+	notaryServer, err := server.New(
+		8000,
+		[]byte(TestServerCertificate),
+		[]byte(TestServerKey),
+		filepath.Join(tempDir, "db.sqlite3"),
+		"example.com",
+		false,
+		logger,
+		NoneEncryptionBackend,
+		PublicConfig,
+	)
+	if err != nil {
+		t.Fatalf("Couldn't get server: %s", err)
+	}
+	testServer := httptest.NewTLSServer(notaryServer.Handler)
+	t.Cleanup(func() {
+		testServer.Close()
+	})
+	return testServer
+}
+
+func MustPrepareAdminAccount(t *testing.T, ts *httptest.Server) string {
```

Comment:
This will create the so called `testadmin` user when it is called even if a user already exists in the PR, in that case the user created won't be admin. Or am I mistaken?

Reply:
It only differs from the other function by the username and the password and whether it creates an admin user or a normal user depends on how it is called.